core/src/autogluon/core/models/ensemble/bagged_ensemble_model.py

```diff
@@ -411,28 +411,17 @@ def _set_n_repeat_single(self):
         self._bagged_mode = False
 
     def _get_default_fold_fitting_strategy(self):
-        # ray not working properly on macos: https://github.com/ray-project/ray/issues/20084
-        # TODO: re-enable macos once this issue is addressed
-        os_fitting_strategy_map = dict(
-            Darwin='sequential_local',
-            Windows='parallel_local',
-            Linux='parallel_local',
-        )
-        current_os = platform.system()
-        fold_fitting_strategy = os_fitting_strategy_map.get(current_os, 'sequential_local')
-        if fold_fitting_strategy == 'sequential_local':
-            warning_msg = f'\tWill use sequential fold fitting strategy because {current_os} OS does not yet support parallel folding.'
+        try:
+            try_import_ray()
+            fold_fitting_strategy = "parallel_local"
+            if os.environ.get("AG_DISTRIBUTED_MODE", False):
```

Comment:
Don't hard-code this in multiple places. Make a function somewhere like `is_distributed_mode()`

Reply:
refer to later comment for a more refined suggestion